expression/expression.go

```diff
@@ -79,6 +79,9 @@ type Expression interface {
 	// IsCorrelated checks if this expression has correlated key.
 	IsCorrelated() bool
 
+	// ReferTable checks if this expression refers to a table.
+	ReferTable() bool
```

Comment:
Good idea, but how about use `ConstItem` like MySQL?

Reply:
yeah，I will rename it.

Reply:
@qw4990 
In MySQL,

> An expression is constant if it:
>     - refers no tables.
>     - refers no subqueries that refers any tables.
>     - refers no non-deterministic functions.
>     - refers no statement parameters. 

The concept `constant` in MySQL is not the same as TiDB.
It seems that a `basic_const_item` is corresponding to `Constant` in TiDB. 
Can we use a better name without confusing?

Reply:
Is there any difference between `basic_const_item` and `const_item()` in MySQL?

Reply:
@winoros 
```
  /*
    Returns true if this is a simple constant item like an integer, not
    a constant expression. Used in the optimizer to propagate basic constants.
    It is assumed that val_xxx() does not modify the item's state for
    such items. It is also assumed that val_str() can be called with nullptr
    as argument as val_str() will return an internally cached const string.
  */
  virtual bool basic_const_item() const { return false; }
```
The concept of `basic_const_item` is close to `Constant` in TiDB.
And `const_item` is a constant expr, and I guess that it can be evaled immediately